connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorStateInfo.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.connect.runtime.rest.entities;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+public class ConnectorStateInfo {
+
+    private final String name;
+    private final ConnectorState connector;
+    private final List<TaskState> tasks;
+
+    @JsonCreator
+    public ConnectorStateInfo(@JsonProperty("name") String name,
+                              @JsonProperty("connector") ConnectorState connector,
+                              @JsonProperty("tasks") List<TaskState> tasks) {
+        this.name = name;
+        this.connector = connector;
+        this.tasks = tasks;
+    }
+
+    @JsonProperty
+    public String name() {
+        return name;
+    }
+
+    @JsonProperty
+    public ConnectorState connector() {
+        return connector;
+    }
+
+    @JsonProperty
+    public List<TaskState> tasks() {
+        return tasks;
+    }
+
+    public abstract static class AbstractState {
+        private final String state;
+        private final String msg;
+        private final String workerId;
+
+        public AbstractState(String state, String workerId, String msg) {
+            this.state = state;
+            this.workerId = workerId;
+            this.msg = msg;
+        }
+
+        @JsonProperty
+        public String state() {
+            return state;
+        }
+
+        @JsonProperty
+        public String workerId() {
+            return workerId;
+        }
+
+        @JsonProperty
+        @JsonInclude(JsonInclude.Include.NON_EMPTY)
```

Comment:
Is there any value in including `null` if this is empty? In some languages it doesn't really matter since a missing field is treated the same anyway, but in others omitting fields might require adding more checks. (Then again, you'll have to check for null anyway, so maybe it doesn't matter...)